pkg/users/account-details.js

```diff
@@ -100,17 +100,8 @@ export function AccountDetails({ accounts, groups, current_user, user, shells })
     const [expiration, setExpiration] = useState(null);
     const [lastlogpath, setLastlogPath] = useState(null);
 
-    useEffect(() => {
-        cockpit.spawn(["test", "-e", "/var/run/utmp"], { err: "ignore" }).then(() => {
-            setLastlogPath("/var/run/utmp");
-        }).catch(() => {
-            cockpit.spawn(["test", "-e", "/var/lib/lastlog/lastlog2.db"], { err: "ignore" }).then(() => {
-                setLastlogPath("/var/lib/lastlog/lastlog2.db");
-            }).catch(() => {
-                setLastlogPath(null);
-            });
-        });
-    }, []);
+    // react wants sync functions that wrap async ones to prevent race conditions
```

Comment:
Oho! I have to read up on this.

Reply:
Aha! The issue is explained here: https://devtrium.com/posts/async-functions-useeffect

In short: The function passed to `useEffect` is expected to return another function which is called as cleanup. However, calling a `async` function (without await) returns a Promise, which confuses useEffect.

So, `useEffect(() => async () => ...)` will actually treat the async function as the cleanup function, which will also not work, I guess. The tests should show this, let's see.

Sorry for sending us down this rabbit hole, I have in fact never used a async function with `useEffect` before. I prefer our own `useInit` or `useObject` hooks, which are more straightforward with cleanup functions.

So, we could write this as

```
useInit(async () => setLastlogPath(await getLastlogPath()));
```

or keep `useEffect` but with Promises

```
useEffect(() => {
   getLastLogPath().then(setLastLogPath);
}
```

I prefer `useInit`, so I vote for that.

Reply:
Thanks! The comment can be removed now, no?

Reply:
I have removed it, let's land this when the tests are green enough.